modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js

```diff
@@ -181,10 +181,11 @@ export default function Chart({
 				? HOUR_IN_MILLISECONDS
 				: DAY_IN_MILLISECONDS;
 
-		const keys = [
-			'analyticsReportsHistoricalViews',
-			'analyticsReportsHistoricalReads',
-		];
+		const keys = new Set(['analyticsReportsHistoricalViews']);
+
+		if (dataProviders.length === 2) {
+			keys.add('analyticsReportsHistoricalReads');
+		}
```

Comment:
Also, while removing the `dataProvider` attribute (see the other comment https://github.com/liferay-frontend/liferay-portal/pull/885#discussion_r591457195), I can create a `Map` with `{enpoint : key}` instead of hard-coding it